homeassistant/components/airzone/sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Support for the Airzone sensors."""
+from __future__ import annotations
+
+from typing import Final
+
+from aioairzone.const import AZD_HUMIDITY, AZD_NAME, AZD_TEMP, AZD_ZONES
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AirzoneEntity
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+
+SENSOR_TYPES: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        key=AZD_TEMP,
+        name="Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        device_class=DEVICE_CLASS_HUMIDITY,
+        key=AZD_HUMIDITY,
+        name="Humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        zone_name = zone_data[AZD_NAME]
+
+        for description in SENSOR_TYPES:
+            if description.key in zone_data:
+                sensors.append(
+                    AirzoneSensor(
+                        coordinator,
+                        description,
+                        zone_id,
+                        zone_name,
+                    )
+                )
+
+    async_add_entities(sensors)
+
+
+class AirzoneSensor(AirzoneEntity, SensorEntity):
+    """Define an Airzone sensor."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        description: SensorEntityDescription,
+        zone_id: str,
+        zone_name: str,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, zone_id, zone_name)
+        self._attr_name = f"{zone_name} {description.name}"
+        self._attr_unique_id = f"{zone_id}_{description.key}"
```

Comment:
It doesn't look like `zone_id` is unique if they have multiple systems

Is there a serial number of other acceptable unique id that this can be prefixed with ?
https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique#unique-id-requirements

If not you can use the https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique#unique-id-of-last-resort as a prefix

Reply:
> Is there a serial number of other acceptable unique id that this can be prefixed with ? https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique#unique-id-requirements
No, the API doesn't expose a serial number or any other acceptable unique ID...
Or at least as far as I know, based on the [documentation of the API](https://github.com/Noltari/aioairzone/blob/master/docs/airzone-local-api.pdf)...

> If not you can use the https://developers.home-assistant.io/docs/entity_registry_index?_highlight=unique#unique-id-of-last-resort as a prefix
Ok, I think I'll also use IP & port...

Reply:
@bdraco changed in 1f7df024824d93ded844045b3bdf53d4c4ec9810